### PR TITLE
build: Update go modules to build with go 1.18 and support go 1.19

### PR DESCRIPTION
**Pull request by travisn**
**Description of your changes:**
- Update from make go.mod.update to ensure we are consuming the latest stable packages available.
- Support both go 1.18 and 1.19 and drop 1.17
- Update to gofmt required for 1.19

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure)).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by sp98** [comment]
The darwin amd64 failure in the `Builds` CI suggests [updating](https://github.com/golang/go/issues/51706#issuecomment-1074480464) `golang.org/x/sys` version

**Comment by BlaineEXE** [comment]
Also this stackoverflow: https://stackoverflow.com/questions/71507321/go-1-18-build-error-on-mac-unix-syscall-darwin-1-13-go253-golinkname-mus

**Comment by BlaineEXE** [comment]
Are there issues going to controller-tools 0.9.2 directly?

**Comment by travisn** [comment]
> Are there issues going to controller-tools 0.9.2 directly?

0.7.0 or newer gives this error:
```
Error: unable to parse option "crd:maxDescLen=-1,trivialVersions=true,generateEmbeddedObjectMeta=true,allowDangerousTypes=true": [unknown argument "trivialVersions" (at <input>:1:30) extra arguments provided: "true,generateEmbeddedObjectMeta=true,allowDangerousTypes=true" (at <input>:1:31)]
```

So i'd propose we update the controller runtime further in a separate PR.

**Comment by travisn** [comment]
An annoyance of this change is that `make codegen` will produce different results on go 1.18 and 1.19. We need to go with the codegen format from 1.19, otherwise the 1.19 build will fail.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -17,109 +17,114 @@ permissions:
 
 jobs:
   macos-build:
-    runs-on: macos-10.15
+    runs-on: macos-12
```
Comment (BlaineEXE) [inline .github/workflows/build.yml:20]: NICE!

**Review by BlaineEXE — APPROVED** [review]

**Pull request merged**